src/Microsoft.Health.Fhir.Core/Features/Conformance/Models/ListedResourceComponent.cs

```diff
@@ -35,7 +36,8 @@ public ListedResourceComponent()
 
         public string Type { get; set; }
 
-        public string Profile { get; set; }
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "CA2227:Collection properties should be read only", Justification = "Need to set in builder")]
+        public ICollection<string> Profile { get; set; }
```

Comment:
Profile is not an `ICollection`. Perhaps just use your `ReferenceComponent` object